query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -189,6 +369,58 @@ public void handleRequest(
     }
   }
 
+  @Nonnull
+  private QueryRequest rewriteRequestWithViewFiltersApplied(QueryRequest request,
+      Map<String, ViewColumnFilter> columnFilterMap) {
+
+    Filter newFilter = removeViewColumnFilter(request.getFilter(), columnFilterMap);
+    QueryRequest.Builder builder = QueryRequest.newBuilder(request).clearFilter();
+    if (!Filter.getDefaultInstance().equals(newFilter)) {
+      builder.setFilter(newFilter);
+    }
+    return builder.build();
+  }
+
+  private Filter removeViewColumnFilter(Filter filter,
+      Map<String, ViewColumnFilter> columnFilterMap) {
+    if (filter.getChildFilterCount() > 0) {
+      // Recursively try to remove the filter and eliminate the null nodes.
+      Set<Filter> newFilters = filter.getChildFilterList().stream()
+          .map(f -> removeViewColumnFilter(f, columnFilterMap))
+          .filter(f -> !Filter.getDefaultInstance().equals(f))
+          .collect(Collectors.toCollection(LinkedHashSet::new));
+
+      if (newFilters.isEmpty()) {
+        return Filter.getDefaultInstance();
+      } else if (newFilters.size() == 1) {
+        return newFilters.stream().findFirst().get();
+      } else {
+        return Filter.newBuilder(filter).clearChildFilter().addAllChildFilter(newFilters).build();
+      }
+    } else {
+      return rewriteLeafFilter(filter, columnFilterMap);
+    }
+  }
+
+  private Filter rewriteLeafFilter(Filter queryFilter,
+      Map<String, ViewColumnFilter> columnFilterMap) {
+    ViewColumnFilter viewColumnFilter =
+        columnFilterMap.get(queryFilter.getLhs().getColumnIdentifier().getColumnName());
+    if (viewColumnFilter == null) {
+      return queryFilter;
+    } else {
+
+      // The only case where we need to return non-null filter is when view filter is based on 'IN'
```

Comment:
are there cases here with different RHS? like:
VF: `X = 'A'`
QF: `X = 'A' AND X ='B'` - should return nothing

VF: `X IN ('A', 'B')`
QF: `X IN ('A', 'B') AND X ='B'` - should return only B matches

Reply:
@buchi-busireddy - this is the last comment I think we need to make sure we resolve (either through comment or fix)

Reply:
You need to get 5 ⭐ for your reviews 👍 

I've fixed this case and added unit test also now.